libafl_bolts/src/rands.rs

```diff
@@ -363,26 +434,26 @@ impl Rand for XkcdRand {
 /// A test rng that will return the same value (chose by fair dice roll) for testing.
 impl XkcdRand {
     /// Creates a new [`XkcdRand`] with the rand of 4, [chosen by fair dice roll, guaranteed to be random](https://xkcd.com/221/).
-    /// Will always return this seed.
     #[must_use]
     pub fn new() -> Self {
-        Self::with_seed(4)
+        Self { val: 4 }
     }
 
-    /// Creates a new [`XkcdRand`] with the given seed. Will always return this seed.
+    /// Creates a new [`XkcdRand`] with the given seed.
     #[must_use]
     pub fn with_seed(seed: u64) -> Self {
-        Self { val: seed }
+        let mut rand = XkcdRand { val: 0 };
```

Comment:
We usually use `Self` in this case. Also, I don't get the reason for this specific change

Reply:
The reason for this change is to ensure that regardless of the seeding, the output of `next_float()` or `below()` is reasonable (and changes with changes to the seed). With direct seeding without `splitmix` e.g. `below(2^32)` will return 0 for all 32-bit bit seeds.

Reply:
Won't the output of `next_float` and `below` always be the same for XkcdRand anyway?

Reply:
`XkcdRand` is a joke type... https://xkcd.com/221/